### DISCUSSION TITLE
How to resolve Error: The file "/vercel/path0/.next/routes-manifest.json" couldn't be found.

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#47517] [Help]
### Summary

I have been getting the following on Vercel builds/deployments continuously after somewhat halfway potentially resolving [this other Vercel error](https://github.com/vercel/next.js/discussions/47455):

```
(node:237) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
--
00:07:34.881 | (Use `node --trace-warnings ...` to show where the warning was created)
00:07:35.842 | info  - Linting and checking validity of types...
00:09:43.572 | Error: The file "/vercel/path0/.next/routes-manifest.json" couldn't be found. This is often caused by a misconfiguration in your project.
00:09:43.572 | Learn More: https://err.sh/vercel/vercel/now-next-routes-manifest
00:09:43.828 | NEXT_NO_ROUTES_MANIFEST: The file "/vercel/path0/.next/routes-manifest.json" couldn't be found. This is often caused by a misconfiguration in your project.
```

When I visit https://err.sh/vercel/vercel/now-next-routes-manifest, it talks about making sure your project settings are the default in Next.js on Vercel, which mine are (I double and triple checked).

<img width="953" alt="Screenshot 2023-03-25 at 12 23 13 AM" src="https://user-images.githubusercontent.com/28718/227703368-617a5929-bba3-4bc1-b2a3-901ef2e4a1ea.png">

Locally, I am able to run `npm run lint` with no errors, and `npm run build` works fine. I can develop my app fine as well with `npm run dev`.

I was facing [that other hanging error](https://github.com/vercel/next.js/discussions/47455) before, but after removing most of the calls to loading JSON modules in the `getServerSideProps`, now I am getting this above error... Any ideas how I can fix this or what else might be happening?

Thank you.

P.S. Why am I not getting the same build response locally on my MacOS M2? Do I need to resort to simulating this somehow locally as a linux machine? If so, how can I do that do you think?

**Comment by lancejpollard** [comment]
One thing that seems weird, it says this in Vercel builds:

```
Detected `package-lock.json` generated by npm 7+...
```

But when I do `npm -v` locally, I get 8.19.3. What should I do?

**Reply by lancejpollard** [reply]
Tried switching to yarn, but same error:

```
Running build in Cleveland, USA (East) – cle1
--
00:34:00.730 | Cloning github.com/... (Branch: main, Commit: c92189b)
00:34:08.745 | Cloning completed: 8.015s
00:34:10.009 | Skipping build cache since Package Manager changed from "npm" to "yarn"
00:34:11.443 | Running "vercel build"
00:34:12.006 | Vercel CLI 28.18.1
00:34:12.460 | Installing dependencies...
00:34:12.837 | yarn install v1.22.17
00:34:12.933 | [1/4] Resolving packages...
00:34:13.402 | [2/4] Fetching packages...
00:34:49.437 | [3/4] Linking dependencies...
00:34:49.439 | warning " > @mdx-js/loader@2.3.0" has unmet peer dependency "webpack@>=4".
00:34:49.441 | warning " > raw-loader@4.0.2" has unmet peer dependency "webpack@^4.0.0 \|\| ^5.0.0".
00:34:49.442 | warning " > styled-components@5.3.9" has unmet peer dependency "react-is@>= 16.8.0".
00:35:02.165 | [4/4] Building fresh packages...
00:35:10.485 | Done in 57.65s.
00:35:10.519 | Detected Next.js version: 13.2.4
00:35:10.522 | Running "yarn run build"
00:35:10.740 | yarn run v1.22.17
00:35:10.785 | $ next build
00:35:11.431 | (node:410) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
00:35:11.431 | (Use `node --trace-warnings ...` to show where the warning was created)
00:35:12.216 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
00:35:12.217 | This information is used to shape Next.js' roadmap and prioritize features.
00:35:12.217 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
00:35:12.217 | https://nextjs.org/telemetry
00:35:12.217 |  
00:35:12.416 | info  - Linting and checking validity of types...
```

Then:

```
Error: The file "/vercel/path1/.next/routes-manifest.json" couldn't be found. This is often caused by a misconfiguration in your project.
--
00:36:43.359 | Learn More: https://err.sh/vercel/vercel/now-next-routes-manifest
00:36:43.616 | NEXT_NO_ROUTES_MANIFEST: The file "/vercel/path1/.next/routes-manifest.json" couldn't be found. This is often caused by a misconfiguration in your project.
```

**Comment by adelindanaila** [comment]
Same issue here, turborepo environment, the error seems to appear only when the preview -> production migration is happening.
After redeploying without build cache, everything is back to normal.

**Reply by lancejpollard** [reply]
Mine seemed to have resolved with me just removing all the pages folder, and progressively adding them back, something just caused it to go away and I'm not sure what it was :/

**Reply by anthonyshew** [reply]
I'm a little late but for anyone that ends up here after searching:

If you're seeing cache hits with `turbo` and things aren't being found in your build, you likely have misconfigured [your `outputs` array](https://turbo.build/repo/docs/reference/configuration#outputs). If you don't have an `outputs` array, Turborepo is unable to cache your build outputs, so a cache hit would result in restoring _nothing_ from the cache.

**Reply by pantharshit007** [reply]
hey, @anthonyshew 
it was working fine up until but it started causing this issue after I updated my .github/build.yml file I know it doesn't have any relation with it but that's the error I am receiving re-deploying solves it but I am getting it on every commit from then onwards
this is my outputs array
```
"build": {
      "dependsOn": ["^build"],
      "inputs": ["$TURBO_DEFAULT$", ".env*"],
      "outputs": [".next/**", "!.next/cache/**"]
    },
```

I can send you the repo link if you could check the `turbo.json`

EDIT: I added individual output arrays and it started working fine
```json
    "user-app#build": {
      "dependsOn": ["^build"],
      "env": ["AUTH_*", "RESEND_API_KEY"],
      "outputs": [".next/**", "!.next/cache/**"]
    },
```

**Comment by DJanocha** [comment]
In my case I got that error when trying to deploy a next.js project (called 'nextjs') inside of monorepo but didn't change the root directory  to 'apps/nextjs'. So my problem was not about the commands overriding (build/install or whatever).

**Reply by giorgiabosello** [reply]
Thank you! Same!

**Comment by AliWayne101** [comment]
In my case i just removed `distDir: 'build'` from next.config.js

**Reply by Syed-Hasnain-Askari** [reply]
Thankyou so much!

**Reply by dew97-tech** [reply]
Thanks a lot !

**Reply by Ida-Naliaka** [reply]
Thank you! worked for me

**Reply by tom-leamon** [reply]
Thank you! I totally forgot I had set that

**Reply by rkamely** [reply]
you are amazing thanks a lot

**Reply by MostafaS** [reply]
thanks, worked for me

**Reply by Shaybush** [reply]
hell yeah, working !

**Reply by KhizerWHussain** [reply]
Your Amazing, it works
May Allah bless you.

**Reply by athkdev** [reply]
thank you so much! worked for me

**Reply by Ali3905** [reply]
> In my case i just removed `distDir: 'build'` from next.config.js

work for me

**Reply by waloZarate** [reply]
Thank you, it works for me too

**Reply by rogerwillis** [reply]
Sweet! That's what did it for me as well!

**Reply by umkl** [reply]
thank you, was the issue for me as well, I wonder where vercel normally expects the build to be

**Reply by AhmedSamyMahmoud** [reply]
works for me

**Comment by Kinfe123** [comment]
Hello there , mine is not working... i have done every step that you have suggested but still not working , the web is built using pnpm + turbo + nextjs

**Reply by Himanshuwadhwa1** [reply]
same here, if you found any possible solution then please help!

**Comment by kkcy** [comment]
I have the same issue, nextjs project with yarn. But it can be solve by clicking `redeploy` every time. Which is quite a hassle

**Comment by n2-freevas** [comment]
If we specify `distDir` in `next.config.js`, 
You shoud input same name in Vercel > Build & Deployment Setting > Next.js > Output Directory.
(I tend to forget too.)

**Reply by jokereven** [reply]
![image](https://github.com/vercel/next.js/assets/73232691/318065d7-f6d3-46e4-b5e7-4e8f0fe5f54d)
that is my config, thanks.

**Reply by basedfund** [reply]
This helped, thanks!

**Reply by lalshubham** [reply]
Thanks @n2-freevas, it fixed my deployment error.

**Reply by NgeNgeSoe** [reply]
set value in install command fixed my error.

**Comment by thomasdavis** [comment]
I am in a turbo setup too, and mine has started triggering this error. 

I `checkout` the last commit to work, made a new branch, made a change to the root README.md and pushed the branch.

It still triggers this error, I cannot figure out why my deployment is borked. I think it is something to do with Vercel.

Edit: I got mine working after 100 or so builds of doing stuff, not sure if this fixed it, but I tested changing this to and back, and it only succeeds when I use double quotes in my `next.config.js` 

<img width="804" alt="Screen Shot 2023-06-22 at 11 53 17 pm" src="https://github.com/vercel/next.js/assets/416209/e1e00574-925b-4906-997e-127de140e558">

Good luck everyone else lol

**Reply by rlancejohnson** [reply]
This worked for us, thanks!

**Comment by jpvalery** [comment]
I had this error on a new folder/project in a turborepo setup and I found out that somehow Vercel was overriding the build command with something wrong (like `cd` to the root folder and then building in the wrong subfolder (which was the first folder in that projects folder)).

I disabled the override and it worked immediately
<img width="908" alt="image" src="https://github.com/vercel/next.js/assets/4112343/cb7b3823-08b7-4e9f-94ce-999bc3c088a7">

Hope this helps!

**Reply by nandorojo** [reply]
Thank you! Removing the override script where I called Turbo myself I think fixed it.

**Reply by cred33** [reply]
this worked beautifully! thank you

**Comment by MiguelGGMM** [comment]
I removed `output: 'export'` from `next.config.js` and finally worked! for some reason build files were generated in `.next` folder instead of `build` for some reason, even when I had set `distDir: build`

**Comment by alexboii** [comment]
same issue here :/ no answer from vercel support of course. this is severely impacting our engineering team's productivity

**Comment by polooner** [comment]
I encountered the same issue, I am using a monorepo structure with Turborepo. What worked for me is overriding my deployment/build commands on Vercel to these:
I have a basic workspaces setup:

apps/**
packages/**

![Screenshot 2023-10-13 at 2 35 37 PM](https://github.com/vercel/next.js/assets/114031148/e2d29617-8f59-471b-b379-99111e401bba)

**Reply by polooner** [reply]
And pnpm as shown.

**Reply by joshy36** [reply]
Super helpful, thanks. FYI, all i needed to do was change my output directory. Im using npm.

**Reply by eyaleizenberg** [reply]
Worked for me as well (npm)

**Reply by ace3** [reply]
thats it's helped me also.

**Reply by Emanuel-Manrique** [reply]
Worked for me

**Comment by ludwighogstrom** [comment]
I had the same problem. It looks like it went away when I made sure all `process.env` variables used in the project were listed in the turbo file. See https://turbo.build/repo/docs/core-concepts/caching/environment-variable-inputs

The ESLint plugin `eslint-config-turbo` helped figure out which ones (https://turbo.build/blog/turbo-1-4-0#eslint-config-turbo).

**Comment by rlancejohnson** [comment]
Have any of these changes actually fixed the problem for good? It seems like the issue will be resolved for us for a week or two and then show up again.

**Reply by ludwighogstrom** [reply]
My problem hasn't come back. See above about listing envs in turbo file https://github.com/vercel/next.js/discussions/47517#discussioncomment-7404502

If the problem comes and goes it could have something to do with caching of the build? And that not having all used envs in turbo file makes the cache invalidation problematic. 

Just guessing here though...

**Comment by Barbeariafacil** [comment]
I resolved it by disabling the folder and path modules.
![Captura de Tela 2024-01-17 às 18 30 20](https://github.com/vercel/next.js/assets/156945799/e3659b3f-59a1-4f6c-98d6-065a11170c12)

**Comment by kkcoms** [comment]
Sometimes it helps

if you add ` /.next`

<img width="971" alt="image" src="https://github.com/vercel/next.js/assets/14144728/1639881f-2850-4800-845d-15855debeef6">

<img width="459" alt="image" src="https://github.com/vercel/next.js/assets/14144728/98bcb5f7-fc6a-4eb5-bcf7-3a1dcbaf7d07">

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  images: {
    domains: [
      "files.edgestore.dev"
    ]
  }
}

module.exports = nextConfig
```

<img width="426" alt="image" src="https://github.com/vercel/next.js/assets/14144728/3e3d38ee-5fde-401d-93f2-19e8fdcdead4">

**Comment by nateshirley** [comment]
I ran into "vercel/path0: file not found" errors in a monorepo and fixed it by turning off the build cache. Set env var VERCEL_FORCE_NO_BUILD_CACHE=1 in project settings

**Comment by myyellowshoe** [comment]
Adding my commentary here, after a looot of playing around, adding the correct output directory seamed to fix it for us. I used `vercel.json` because we needed be able to test in a branch without affecting main config settings.

```
{
  "outputDirectory": "apps/web/.next"
}
```

Here's my turbo.json as well

```
{
  "$schema": "https://turbo.build/schema.json",
  "globalDependencies": ["**/.env.*local"],
  "globalEnv": [
    "API_KEYS"
  ],
  "tasks": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".next/**", "!.next/cache/**"]
    },
    "lint": {},
    "dev": {
      "cache": false,
      "persistent": false
    }
  }
}
```

**Reply by MikeCodeur** [reply]
```
{
  "outputDirectory": "apps/xxx/.next"
}
```
works for me

**Comment by levivoelz** [comment]
I had this problem because I started a turbo project without Next.js and didn't have `[".next/**", "!.next/cache/**"]` in `"outputs"`.

before
```json
"build": {
      "dependsOn": ["^build"],
      "inputs": ["$TURBO_DEFAULT$", ".env*"],
      "outputs": ["dist/**", "target/**"]
    },
```

after
```json
"build": {
      "dependsOn": ["^build"],
      "inputs": ["$TURBO_DEFAULT$", ".env*"],
      "outputs": ["dist/**", "target/**", ".next/**", "!.next/cache/**"]
    },
```

**Comment by Xunnamius** [comment]
Had this problem, just solved it. Turns out my Next.js config file wasn't even being read!

Next.js is cool with `next.config.mjs` files when the root `package.json` file has type "commonjs", but Vercel's build environment is not. Switching to `next.config.js` fixed this issue for me, which was because my custom `distDir` setting wasn't getting picked up (along with the rest of the config file).

**Comment by azeezat** [comment]
The following worked for me. I am using an nx monorepo

**On vercel**

Your root directory should be apps/app-api (this is because I am using a monorepo)


<img width="996" height="396" alt="Screenshot 2025-07-19 at 12 55 58 AM" src="https://github.com/user-attachments/assets/07d36f4c-e809-4c78-bd32-255ae4840ca0" />


**Inside `next.config.js`**

```js
const nextConfig = {
  // Use this to set Nx-specific options
  // See: https://nx.dev/recipes/next/next-config-setup
  nx: {},
  outputDirectory: 'apps/app-api/.next',
};
```

**Comment by bombillazo** [comment]
In our case, we were getting a doubled path:

```
Error: ENOENT: no such file or directory, lstat '/vercel/path0/vercel/path0/.next/routes-manifest.json'
```

To fix the issue, we had to remove `outputFileTracingRoot` from our `next.config.js` file to make it work.

**Comment by Tureckiy-zart** [comment]
After removing or commenting out outputFileTracingRoot: path.join(__dirname, '../') in my next.config.js, the build succeeded without the ENOENT: ... routes-manifest.json error. This fix solved the issue for me. Thanks!!!

**Comment by alsongard** [comment]
Had the same issue: ENOENT: no such file or directory, lstat '/vercel/path0/.next/routes-manifest-deterministic.json'
for me it was a configuration on settings for my project on vercel :
Setting: Include files outside the root directory in the [Build Step](https://vercel.com/docs/deployments/build-step). 
Disable  this.
Thanks @lancejpollard
